vllm/entrypoints/openai/api_server.py

```diff
@@ -513,6 +513,31 @@ async def get_tokenizer_info(raw_request: Request):
             )
 
 
+def maybe_register_profiler_endpoints(args):
+    """Conditionally add profiler start/stop endpoints if enabled."""
+    profiler_config = args.profiler_config
+    assert isinstance(profiler_config, ProfilerConfig)
+    if profiler_config.profiler != "none":
+        logger.warning_once(
+            f"Profiler with mode '{profiler_config.profiler}' is enabled in the "
+            "API server. This should ONLY be used for local development!"
+        )
```

Comment:
Logging with f-string

```suggestion
        logger.warning_once(
            "Profiler with mode '%s' is enabled in the "
            "API server. This should ONLY be used for local development!", 
            profiler_config.profiler
        )
```

Reply:
It's a performance thing.

With f-strings the substitution will always be evaluated, even if the statement is never logged (i.e. hidden behind `if` or log level is set to >warning in this case)

See https://docs.astral.sh/ruff/rules/logging-f-string/ for a full explanation